master/master.go

```diff
@@ -16,28 +16,30 @@ import (
 type mst struct {
 	filePath        string             // a path to a raw sequencing FASTQ file to perform assembly on
 	outPath         string             // a path to the location where results will be stored
+	numThreads      int                // number of threads to use for slave processes
 	assemblyResults chan result.Result // a collection of assembly results used by the assembly slave
 	parsingResults  chan result.Result // a collection of parsing results used by the parsing slave
 }
 
 // New creates and returns a new master struct for the file located at the given file path
-func New(rsf, out string) Master {
+func New(rsf, out string, numThreads int) Master {
 	return &mst{
 		filePath:        rsf,
 		outPath:         out,
+		numThreads:      numThreads,
 		assemblyResults: make(chan result.Result),
 		parsingResults:  make(chan result.Result),
 	}
 }
 
 // Process launches the assembly of the contigs it was created with
 func (m *mst) Process() error {
-	assemblySlave := slave.New("assembly process initiated by master", m.filePath, m.outPath, slave.Assembly)
+	assemblySlave := slave.New("assembly process initiated by master", m.filePath, m.outPath, m.numThreads, slave.Assembly)
 	if _, err := assemblySlave.Process(); err != nil {
 		return fmt.Errorf("slave assembly process failed with err: %v", err)
 	}
 
-	parserSlave := slave.New("reporting/parse process initiated by master", m.filePath, m.outPath, slave.Parse)
+	parserSlave := slave.New("reporting/parse process initiated by master", m.filePath, m.outPath, m.numThreads, slave.Parse)
```

Comment:
The number of parameters keeps increasing and I do not know whether we want to keep mixing params since the slave interface is consistent between assembly and parsing. Let's chat about this one. We can change the interface to take in a parameter struct that indicates what slave to run and take in the respective parameters.